### PR TITLE
TE5-T422 ListView and Bottom Navigation

### PR DESCRIPTION
**Pull request by jay-a-shah**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,23 +11,83 @@
...
         android:theme="@style/Theme.DemoAppkotlin">
         <activity
-            android:name=".differentLayout.LinearLayoutActivity"
+            android:name=".listView_ViewPager.ExpandableRecyclerViewActivity"
+            android:label="Expandable Recycler View "
+            android:parentActivityName=".listView_ViewPager.ListViewActivity"
+            android:exported="false" />
+        <activity
+            android:name=".listView_ViewPager.ExpandableListViewActivity"
+            android:label="Expandable List View"
```
Comment (purvesh-d) [inline app/src/main/AndroidManifest.xml]: move to string file, in whole files

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import android.widget.CheckBox
+import android.widget.ImageView
+import android.widget.TextView
+import com.example.demoappkotlin.R
+import androidx.recyclerview.widget.RecyclerView
+import com.example.demoappkotlin.listView_ViewPager.ModelClasses.ModelRecycle
+
+class RecycleAdapter(var list: ArrayList<ModelRecycle>) :
+    RecyclerView.Adapter<RecycleAdapter.RecycleViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecycleViewHolder {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt:16]: put empty line

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        val itemView =
+            LayoutInflater.from(parent.context).inflate(R.layout.item_recycler_view, parent, false)
+        return RecycleViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: RecycleViewHolder, position: Int) {
+        val listData = list[position]
+        holder.setIsRecyclable(false)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt]: use apply

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import android.os.Bundle
+import androidx.viewpager.widget.ViewPager
+import com.example.demoappkotlin.R
+import com.example.demoappkotlin.listView_ViewPager.Adapters.BottomNavigationAdapter
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
+class BottomNavigationActivity : AppCompatActivity() {
+    lateinit var bottomNavigationView: BottomNavigationView
+    lateinit var bottomNav: ViewPager
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt:15]: put extra empty line

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        bottomNavigationView = findViewById<BottomNavigationView>(R.id.bottom_navigation)
+        bottomNav = findViewById(R.id.viewPagerBottomNavigation)
+        setViewPagerAdapter()
+        setBottomNavigation()
+    }
+
+    private fun setBottomNavigation() {
+        bottomNavigationView.setOnNavigationItemSelectedListener {
+            when (it.itemId) {
+                R.id.icon_home -> bottomNav.currentItem = 0
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt]: static integer

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/CustomListViewActivity.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import android.os.Bundle
+import android.widget.ListView
+import android.widget.Toast
+import com.example.demoappkotlin.R
+import com.example.demoappkotlin.listView_ViewPager.Adapters.MyAdapter
+import com.example.demoappkotlin.listView_ViewPager.ModelClasses.ModelList
+
+class CustomListViewActivity : AppCompatActivity() {
+    lateinit var myAdapter: MyAdapter
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/CustomListViewActivity.kt:15]: put empty line before start of fun in all the files

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/FirstFragment.kt [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package com.example.demoappkotlin.listView_ViewPager.Fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.demoappkotlin.R
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/FirstFragment.kt:9]: extra empty line

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/FirstFragment.kt [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+class FirstFragment : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/FirstFragment.kt]: why this is empty block of let?

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/MyFavourites.kt [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import android.view.ViewGroup
+import com.example.demoappkotlin.R
+
+
+class MyFavourites : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/MyFavourites.kt]: why empty block?

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/Recent.kt [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import android.view.ViewGroup
+import com.example.demoappkotlin.R
+
+
+class Recent : Fragment() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+        }
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/Recent.kt]: same

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/Recent.kt [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Fragments/Recent.kt]: code formatting

File: app/src/main/java/com/example/demoappkotlin/utils/Constant.kt [hunk 12]
```diff
@@ -1,2 +1,5 @@
 package com.example.demoappkotlin.utils
-const val Zero = 0
\ No newline at end of file
+const val Zero = 0
+const val One = 1
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/utils/Constant.kt]: use only capital word for naming variables

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt (hunk 4 above)
Comment (jay-a-shah) [reply app/src/main/java/com/example/demoappkotlin/listView_ViewPager/BottomNavigationActivity.kt:15]: Okk

File: app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt [hunk 13]
```diff
@@ -17,12 +19,13 @@ class BasicsKotlin : AppCompatActivity() {
             showToast()
         }
     }
+
     fun showToast() {
         val toastView = layoutInflater.inflate(
-                R.layout.custom_toast, findViewById(R.id.customToastLayout))
+            R.layout.custom_toast, findViewById(R.id.customToastLayout))
         val myToast = Toast(applicationContext)
         myToast.duration = Toast.LENGTH_SHORT
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt]: use apply

File: app/src/main/java/com/example/demoappkotlin/MainActivity.kt [hunk 14]
```diff
@@ -25,10 +27,10 @@ class MainActivity : AppCompatActivity() {
         val androiduibtn = findViewById<Button>(R.id.btnAndroidUi)
         androiduibtn.setOnClickListener {
             val normalToast = Toast.makeText(
-                    applicationContext, getString(R.string.toast_normal),
-                    Toast.LENGTH_SHORT
+                applicationContext, getString(R.string.toast_normal),
+                Toast.LENGTH_SHORT
             )
-            normalToast.setGravity(Gravity.TOP, Zero, Zero)
+            normalToast.setGravity(Gravity.TOP, zero, zero)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/MainActivity.kt]: use apply

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/RecyclerViewActivity.kt [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+            add(ModelRecycle("Kotlin", R.drawable.kotlin, false))
+            add(ModelRecycle("larvel", R.drawable.larvel, false))
+            add(ModelRecycle("Android", R.drawable.android, false))
+            add(ModelRecycle("Kotlin", R.drawable.kotlin, false))
+            add(ModelRecycle("larvel", R.drawable.larvel, false))
+            add(ModelRecycle("Android", R.drawable.android, false))
+            add(ModelRecycle("Kotlin", R.drawable.kotlin, false))
+            add(ModelRecycle("larvel", R.drawable.larvel, false))
+        }
+        recyclerview.addItemDecoration(DividerItemDecoration(baseContext,
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/RecyclerViewActivity.kt]: use apply

File: app/src/main/res/layout/activity_bottom_navigation.xml [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+        android:id="@+id/viewPagerBottomNavigation"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toTopOf="@id/bottom_navigation"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <com.google.android.material.bottomnavigation.BottomNavigationView
+        android:id="@+id/bottom_navigation"
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_bottom_navigation.xml]: naming

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt [hunk 17]
```diff
@@ -1,12 +1,14 @@
 package com.example.demoappkotlin
+
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.view.Gravity
 import android.widget.Button
 import android.widget.Toast
-import com.example.demoappkotlin.utils.Zero
+import com.example.demoappkotlin.utils.zero
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/BasicsKotlin.kt]: constant file should contains only capitals word, `const val ZERO = 0`

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/BottomNavigationAdapter.kt [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+
+class BottomNavigationAdapter(var fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
+
+    override fun getCount(): Int {
+        return three
+    }
+
+    override fun getItem(position: Int): Fragment {
+        return when (position) {
+            zero -> Recent()
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/BottomNavigationAdapter.kt]: make full capital in constant, not small

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+        return RecycleViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: RecycleViewHolder, position: Int) {
+        val listData = list[position]
+        holder.apply {
+            setIsRecyclable(false)
+            txtView?.text = listData.name
+            imgView?.setImageResource(listData.image)
+            chkbxView?.isChecked = listData.check
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt]: use apply

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/ListViewActivity.kt [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+        btnListView.setOnClickListener {
+            val intent = Intent(applicationContext, SimpleListViewActivity::class.java)
+            startActivity(intent)
+        }
+        val btnListViewArray = findViewById<Button>(R.id.btnListViewTwo)
+        btnListViewArray.setOnClickListener {
+            val intent = Intent(applicationContext, SimpleListViewWithArrayActivity::class.java)
+            startActivity(intent)
+        }
+        val btnCustomListView = findViewById<Button>(R.id.btnListViewThree)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/ListViewActivity.kt:25]: code formating

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/MyAdapter.kt [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.ImageView
+import android.widget.TextView
+import com.example.demoappkotlin.R
+import com.example.demoappkotlin.listView_ViewPager.ModelClasses.ModelList
+import com.example.demoappkotlin.utils.ZERO
+
+class MyAdapter(val context: Context, val list: ArrayList<ModelList>) : BaseAdapter() {
+    private var ContactList = arrayListOf<ModelList>()
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/MyAdapter.kt]: variable name must start with small case

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+    override fun onBindViewHolder(holder: RecycleViewHolder, position: Int) {
+        val listData = list[position]
+        holder.apply {
+            setIsRecyclable(false)
+            txtView?.text = listData.name
+            imgView?.setImageResource(listData.image)
+            chkbxView?.apply {
+                isChecked = listData.check
+                setOnCheckedChangeListener { btn, bool ->
+                    holder.chkbxView?.isChecked = bool
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/Adapters/RecycleAdapter.kt]: `holder.chkbxView?.isChecked = bool` this line is necessary?  try to run without it and if required then utilize apply property.

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/CustomListViewActivity.kt [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+        list.apply {
+            add(ModelList("Android", "A", R.drawable.android))
+            add(ModelList("Kotlin", "k", R.drawable.kotlin))
+            add(ModelList("Swift", "S", R.drawable.swift))
+            add(ModelList("React", "R", R.drawable.larvel))
+            add(ModelList("Mongo", "M", R.drawable.mongo))
+        }
+        val listview = findViewById<ListView>(R.id.customlistview)
+        listview.adapter = myAdapter
+        var new = myAdapter.getItemId(2)
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/CustomListViewActivity.kt]: static integer

File: app/src/main/java/com/example/demoappkotlin/listView_ViewPager/SimpleListViewWithArrayActivity.kt [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+import android.widget.ArrayAdapter
+import android.widget.ListView
+import com.example.demoappkotlin.R
+
+class SimpleListViewWithArrayActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_list_view_with_array)
+        val arrayAdapter:ArrayAdapter<*>
```
Comment (purvesh-d) [inline app/src/main/java/com/example/demoappkotlin/listView_ViewPager/SimpleListViewWithArrayActivity.kt]: code formatting

File: app/src/main/res/layout/activity_bottom_navigation.xml [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+        android:layout_height="wrap_content"
+        android:background="@color/teal_700"
+        android:elevation="8dp"
+        app:itemIconTint="@color/white"
+        app:itemTextAppearanceActive="@style/TextAppearance.MaterialComponents.Headline6"
+        app:itemTextColor="@color/white"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:menu="@menu/bottom_navigation" />
```
Comment (purvesh-d) [inline app/src/main/res/layout/activity_bottom_navigation.xml]: menu layout name should start with `menu` keyword. Like `@menu/menu_bottom_navigation`

File: app/src/main/res/values/strings.xml [hunk 26]
```diff
@@ -42,4 +42,95 @@
...
+        <item name="fruits">Olive</item>
+        <item name="fruits">Guava</item>
+        <item name="fruits">Pappaya</item>
+        <item name="fruits">Mosambi</item>
+        <item name="fruits">green Apple</item>
+        <item name="fruits">WaterMelon</item>
+        <item name="fruits">Grapes</item>
+        <item name="fruits">Jack Fruit</item>
+    </string-array>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (purvesh-d) [inline app/src/main/res/values/strings.xml]: remove unnecessary comment

**Review by purvesh-d — CHANGES_REQUESTED** [review]

**Pull request merged**